parsec/core/gui/files_widget.py

```diff
@@ -70,6 +70,101 @@ async def _do_delete(workspace_fs, files, silent=False):
                 raise JobResultError("error", multi=len(files) > 1) from exc
 
 
+async def _do_copy_files(workspace_fs, current_directory, files, source_workspace):
+    last_exc = None
+    error_count = 0
+    for f in files:
+        src = f[0]
+        src_type = f[1]
+        file_name = src.name
+        base_name = pathlib.Path(src.name)
```

Comment:
`base_name` is confusing given it seems to refer to https://docs.python.org/3.8/library/os.path.html#os.path.basename, why not `stem_name` ?

Besides I'm not sure why the conversion to `pathlib.Path` is needed (it gives the impression we are working on the entire path where only the file name is important here)
Last but not least, it seams `pathlib.Path.stem` is fairly simple:
https://github.com/python/cpython/blob/e74fb2d7666eea43ad738528a565bb56bc88c28d/Lib/pathlib.py#L855-L863

So I think we should just make this ourselves:
```python
file_name_with_extensions, *_ = src.name.split(".", 1)
```